src/client/shadows/shadowsshadercallbacks.cpp

```diff
@@ -65,6 +65,11 @@ void ShadowDepthShaderCB::OnSetConstants(
 
 	lightMVP *= driver->getTransform(video::ETS_WORLD);
 
+
+	if (driver->getDriverType() == video::EDT_OGLES2 || driver->getDriverType() == video::EDT_OPENGL3) {
+		auto &texture = driver->getTransform(video::ETS_TEXTURE_0);
+		m_texture.set(texture, services);
```

Comment:
This appears to duplicate something that is (normally) already set in `MainShaderUniformSetter`:
https://github.com/luanti-org/luanti/blob/a0547b34356396e4a2561fc8239c58be64dd8ae8/src/client/shader.cpp#L312-L314